### ISSUE TITLE
[FEATURE]: Consider making `container::checkpoint` behind a feature gate

### ISSUE THREAD

**Issue by Mossaka** [containers/youki#2942]
### Background

As far as I understand, the `checkpoint()` function depends on `rust-criu`, which in turn, depends on `protobuf`. Runwasi does not need the restore/checkpoint feature, and thus there is no need to depend on `protobuf`.  

### Feature Request

I am requesting libcontainer to add a new feature called "checkpoint" and put the checkpoint function behind this feature gate. Additionally make `rust-criu` dependency optional.

**Comment by utam0k** [comment]
What about feature gates that add functionality that is not defined in OCI Runtime?

**Comment by Mossaka** [comment]
Yeah the motivation behind this request is to remove the dependency on `protobuf` as Runwasi does not need checkpoint. I am not sure if I understand your comment on adding feature gates on added funcitonalities.

**Comment by utam0k** [comment]
I think you can probably do it, you can't add a feature just for the checkpoint, but by all means we can do that. 

This means that you can use only the CLI Specification here, and the rest should be a feature like `advanced`. Is it enough?
https://github.com/containers/youki/tree/main/crates/liboci-cli

**Comment by utam0k** [comment]
@Mossaka I'm triaging issues in youki. Do you still need this feature?
